src/ros/ros.ts

```diff
@@ -64,4 +71,27 @@ export interface ROSApi {
     activateRoslaunch: (launchFilepath: string, argument: string) => vscode.Terminal;
 }
 
-export let rosApi: ROSApi = new ros1.ROS1();
+const ros1Api: ROSApi = new ros1.ROS1();
+const ros2Api: ROSApi = new ros2.ROS2();
+const unknownRosApi: ROSApi = new unknownROS.UnknownROS();
+
+export let rosApi: ROSApi = unknownRosApi;
+
+export function selectROSApi(distro: string) {
+    rosApi = unknownRosApi;

```

Comment:
Is there a way to do this without listing the codenames?
Don't forget to add noetic and eloquent

Reply:
Per REP spec, I will swap to use `ROS_VERSION` which should be more resilient when the new distro is coming out.